src/domains/TaskQueue.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 IROHA LAB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A FIFO queue that does not allow duplicate value.
+ */
+export class TaskQueue<T> {
```

Comment:
I'd add some tests for this

Reply:
Yes, there should be some test

Reply:
A test for this has been added. You can check the test in PR check.
BTW, due to security reason, PR need to be marked as Ready To Test to start check.